### PR TITLE
Mini world change: Nightmare Isle + fixes map (houses)

### LINKED ISSUE #814: World Board does not work
The Adventures Store world board is not working and does not have a look function.

### PR DESCRIPTION
**Pull request by dex-89**
**Mini world change list: https://tibia.fandom.com/wiki/Mini_World_Changes**

**Nightmare isle:** https://tibia.fandom.com/wiki/Nightmare_Isles_Mini_World_Change
**Yasir:** https://tibia.fandom.com/wiki/Oriental_Trader_Mini_World_Change
**Fury gates:** https://tibia.fandom.com/wiki/Fury_Gates_Mini_World_Change

- **Added Nightmare isle as mini world change**
- **Activated `world board`**
- **Added item #2070**
- **Fixed houses name**
- **Save guild houses as `guildhall=true` in `houses.xml`**
- **Fixed depotId in towns**
- **Fixed house prices**
- **Missing houses added (they was unregistered)**
- **Existing houses were modified (https://www.tibia.com/community/?subtopic=houses)**

**Thanks for help:** @mrhultin

### CODE REVIEW

File: data/scripts/actions/worldboard.lua [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+local communicates = {
+-- Fury Gates
+[1] = {
+	globalStorage=65000,
+	communicate = "A fiery fury gate has opened near one of the major cities somewhere in Tibia."
+},
+-- Yasir
+[2] = {
+	globalStorage = 65014,
```
Comment (debone) [inline data/scripts/actions/worldboard.lua:11]: `GlobalStorage.Yasir`

**Review by debone — COMMENTED** [review]
RIP world.zip :P

File: data/scripts/actions/worldboard.lua (hunk 1 above)
Comment (dex-89) [reply data/scripts/actions/worldboard.lua:11]: its added to storage.lua if i tried used it error: nil value

**Comment by dex-89** [comment]
> RIP world.zip :P

What do you mean?

**Comment by debone** [comment]
> 
> 
> > RIP world.zip :P
> 
> What do you mean?

Confusing UI in Github led me to think the world.zip was almost gone by accident. I didn't check the file though

**Comment by mrhultin** [comment]
This code is quite messy, why even use a if statement to check what the value of select is?

You have table, config = [], which could just aswell contain the text to print to console.

**Comment by dex-89** [comment]
> This code is quite messy, why even use a if statement to check what the value of select is?
> 
> You have table, config = [], which could just aswell contain the text to print to console.

You're right. But how check `chance` using table? Any idea?

**Comment by mrhultin** [comment]
> 
> 
> > This code is quite messy, why even use a if statement to check what the value of select is?
> > You have table, config = [], which could just aswell contain the text to print to console.
> 
> You're right. But how check `chance` using table? Any idea?

I just did it like this and removed chance variable altogheter.

```
local config = {
	-- ankrahmun - north
	[1] = {
		mapName = "ankrahmun-north",
		trueName = "north of Ankrahmun"
	},
	-- darashia - west
	[2] = {
		mapName = "darashia-west",
		trueName = "west of Darashia"
	},
	-- darashia - north
	[3] = {
		mapName = "darashia-north",
		trueName = "north of Darashia"
	}
}

local NightmareIsle = GlobalEvent("NightmareIsle")
function NightmareIsle.onStartup(interval) 
	local select = config[math.random(#config)]
		
	Game.loadMap('data/world/nightmareisle/' .. select.mapName .. '.otbm')
	print('>> Nightmare Isle will be active '.. select.trueName  ..' today.')
	
	setGlobalStorageValue(GlobalStorage.NightmareIsle, 1)
		
	return true
end
NightmareIsle:register()
```

**Comment by dex-89** [comment]
> > > This code is quite messy, why even use a if statement to check what the value of select is?
> > > You have table, config = [], which could just aswell contain the text to print to console.
> > 
> > 
> > You're right. But how check `chance` using table? Any idea?
> 
> I just did it like this and removed chance variable altogheter.
> 
> ```
> local config = {
> 	-- ankrahmun - north
> 	[1] = {
> 		mapName = "ankrahmun-north",
> 		trueName = "north of Ankrahmun"
> 	},
> 	-- darashia - west
> 	[2] = {
> 		mapName = "darashia-west",
> 		trueName = "west of Darashia"
> 	},
> 	-- darashia - north
> 	[3] = {
> 		mapName = "darashia-north",
> 		trueName = "north of Darashia"
> 	}
> }
> 
> local NightmareIsle = GlobalEvent("NightmareIsle")
> function NightmareIsle.onStartup(interval) 
> 	local select = config[math.random(#config)]
> 		
> 	Game.loadMap('data/world/nightmareisle/' .. select.mapName .. '.otbm')
> 	print('>> Nightmare Isle will be active '.. select.trueName  ..' today.')
> 	
> 	setGlobalStorageValue(GlobalStorage.NightmareIsle, 1)
> 		
> 	return true
> end
> NightmareIsle:register()
> ```

Thank you for help. Reworked.

**Comment by mrhultin** [comment]
> 
> 
> > > > This code is quite messy, why even use a if statement to check what the value of select is?
> > > > You have table, config = [], which could just aswell contain the text to print to console.
> > > 
> > > 
> > > You're right. But how check `chance` using table? Any idea?
> > 
> > 
> > I just did it like this and removed chance variable altogheter.
> > ```
> > local config = {
> > 	-- ankrahmun - north
> > 	[1] = {
> > 		mapName = "ankrahmun-north",
> > 		trueName = "north of Ankrahmun"
> > 	},
> > 	-- darashia - west
> > 	[2] = {
> > 		mapName = "darashia-west",
> > 		trueName = "west of Darashia"
> > 	},
> > 	-- darashia - north
> > 	[3] = {
> > 		mapName = "darashia-north",
> > 		trueName = "north of Darashia"
> > 	}
> > }
> > 
> > local NightmareIsle = GlobalEvent("NightmareIsle")
> > function NightmareIsle.onStartup(interval) 
> > 	local select = config[math.random(#config)]
> > 		
> > 	Game.loadMap('data/world/nightmareisle/' .. select.mapName .. '.otbm')
> > 	print('>> Nightmare Isle will be active '.. select.trueName  ..' today.')
> > 	
> > 	setGlobalStorageValue(GlobalStorage.NightmareIsle, 1)
> > 		
> > 	return true
> > end
> > NightmareIsle:register()
> > ```
> 
> Thank you for help. Reworked.

Awesome stuff man! Really neat rework, I like this stuff and your work alot! <3

**Review by dudantas — APPROVED** [review]
Thanks for your contribution!

**Review by majestyotbr — APPROVED** [review]

**Comment by majestyotbr** [comment]
Thanks for your contribution!

**Pull request merged**